### PR TITLE
KAFKA-15623, Migrate test of Stream module to Junit5 (Stream state)

### PR DESCRIPTION
**Pull request by m1a2st**
https://issues.apache.org/jira/browse/KAFKA-15623
Migrate test of stream module to Junit5

Committer Checklist (excluded from commit message)
 Verify design and implementation
 Verify test coverage and CI build status
 Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java [hunk 1]
```diff
@@ -18,42 +18,87 @@
...
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+public class RocksDBWindowStoreTest {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBWindowStoreTest.java]: Could you please rewrite the parameter by making it be abstract and then add related sub classes? for example:

[RocksDBWindowStoreTest.txt](https://github.com/user-attachments/files/15855803/RocksDBWindowStoreTest.txt)

**Comment by C0urante** [comment]
This doesn't touch on Connect, right? Might want to change the title 😅

**Comment by m1a2st** [comment]
@C0urante, Thanks for your comment, It's my mistake, I change title to stream module.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSessionStoreTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import org.apache.kafka.streams.state.SessionStore;
+import org.apache.kafka.streams.state.Stores;
+
+import static java.time.Duration.ofMillis;
+
+public abstract class AbstractRocksDBSessionStoreTest extends AbstractSessionBytesStoreTest {
+
+    private static final String STORE_NAME = "rocksDB session store";
+
+    public abstract StoreType storeType();
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSessionStoreTest.java]: please remove `public`

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBSessionStoreTest.java [hunk 3]
```diff
@@ -16,78 +16,10 @@
...
-            {StoreType.RocksDBTimeOrderedSessionStoreWithIndex},
-            {StoreType.RocksDBTimeOrderedSessionStoreWithoutIndex}
-        });
-    }
+public class RocksDBSessionStoreTest extends AbstractRocksDBSessionStoreTest {
 
     @Override
-    StoreType getStoreType() {
-        return storeType;
+    public StoreType storeType() {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBSessionStoreTest.java]: please remove `public`

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedSessionStoreWithIndexTest.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+public class RocksDBTimeOrderedSessionStoreWithIndexTest extends AbstractRocksDBSessionStoreTest {
+
+    @Override
+    public StoreType storeType() {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedSessionStoreWithIndexTest.java]: please remove `public`

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedSessionStoreWithoutIndexTest.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+public class RocksDBTimeOrderedSessionStoreWithoutIndexTest extends AbstractRocksDBSessionStoreTest {
+
+    @Override
+    public StoreType storeType() {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedSessionStoreWithoutIndexTest.java]: please remove `public`

File: streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest.java [hunk 6]
```diff
@@ -16,59 +16,37 @@
...
 
-@RunWith(Parameterized.class)
-public class RocksDBTimeOrderedWindowSegmentedBytesStoreTest
+public abstract class AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest
     extends AbstractDualSchemaRocksDBSegmentedBytesStoreTest<KeyValueSegment> {
 
     private static final String METRICS_SCOPE = "metrics-scope";
 
-    private enum SchemaType {
+    protected enum SchemaType {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest.java]: please use package-private

File: streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest.java [hunk 7]
```diff
@@ -16,59 +16,37 @@
...
-            {SchemaType.SessionSchemaWithoutIndex, false}
-        });
-    }
-
-    public RocksDBTimeOrderedWindowSegmentedBytesStoreTest(final SchemaType schemaType, final boolean hasIndex) {
-        this.schemaType = schemaType;
-        this.hasIndex = hasIndex;
-    }
-
+    protected abstract boolean hasIndex();
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest.java]: please use package-private

File: streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest.java [hunk 8]
```diff
@@ -16,59 +16,37 @@
...
+    protected abstract SchemaType schemaType();
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest.java]: please use package-private

File: streams/src/test/java/org/apache/kafka/streams/state/internals/KeyValueStoreBuilderTest.java [hunk 9]
```diff
@@ -47,27 +49,32 @@ public class KeyValueStoreBuilderTest {
...
-        when(supplier.get()).thenReturn(inner);
+    public void setUpWithoutInner() {
         when(supplier.name()).thenReturn("name");
         when(supplier.metricsScope()).thenReturn("metricScope");
         builder = new KeyValueStoreBuilder<>(
-            supplier,
-            Serdes.String(),
-            Serdes.String(),
-            new MockTime()
+                supplier,
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/KeyValueStoreBuilderTest.java:56]: please revert those unrelated changes

File: streams/src/test/java/org/apache/kafka/streams/state/internals/MeteredTimestampedKeyValueStoreTest.java [hunk 10]
```diff
@@ -106,35 +108,37 @@ public class MeteredTimestampedKeyValueStoreTest {
...
-    public void before() {
+    private void setUpWithoutContext() {
         mockTime = new MockTime();
         metered = new MeteredTimestampedKeyValueStore<>(
-            inner,
-            "scope",
-            mockTime,
-            Serdes.String(),
-            new ValueAndTimestampSerde<>(Serdes.String())
+                inner,
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/MeteredTimestampedKeyValueStoreTest.java:114]: please revert those changes

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedSessionSchemaWithoutIndexSegmentedBytesStoreTest.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+public class RocksDBTimeOrderedSessionSchemaWithoutIndexSegmentedBytesStoreTest extends AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest {
+
+    @Override
+    protected boolean hasIndex() {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedSessionSchemaWithoutIndexSegmentedBytesStoreTest.java]: please remove "protected"

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedSessionSchemaWithoutIndexSegmentedBytesStoreTest.java [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+        return false;
+    }
+
+    @Override
+    protected SchemaType schemaType() {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedSessionSchemaWithoutIndexSegmentedBytesStoreTest.java]: please remove "protected"

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedWindowSchemaWithIndexSegmentedBytesStoreTest.java [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+public class RocksDBTimeOrderedWindowSchemaWithIndexSegmentedBytesStoreTest extends AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest {
+
+    @Override
+    protected boolean hasIndex() {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedWindowSchemaWithIndexSegmentedBytesStoreTest.java]: please remove "protected"

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedWindowSchemaWithIndexSegmentedBytesStoreTest.java [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+        return true;
+    }
+
+    @Override
+    protected AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest.SchemaType schemaType() {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedWindowSchemaWithIndexSegmentedBytesStoreTest.java]: please remove "protected"

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedWindowSchemaWithoutIndexSegmentedBytesStoreTest.java [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+
+public class RocksDBTimeOrderedWindowSchemaWithoutIndexSegmentedBytesStoreTest extends AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest {
+
+    @Override
+    protected boolean hasIndex() {
+        return false;
+    }
+
+    @Override
+    protected SchemaType schemaType() {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedWindowSchemaWithoutIndexSegmentedBytesStoreTest.java]: please remove "protected"

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedWindowSchemaWithoutIndexSegmentedBytesStoreTest.java [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+public class RocksDBTimeOrderedWindowSchemaWithoutIndexSegmentedBytesStoreTest extends AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest {
+
+    @Override
+    protected boolean hasIndex() {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedWindowSchemaWithoutIndexSegmentedBytesStoreTest.java]: please remove "protected"

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStoreSegmentValueFormatterTest.java [hunk 17]
```diff
@@ -280,38 +276,27 @@ public void shouldCreateNewWithRecord() {
      * inconsistency among segments. These tests verify that the store is able to recover
      * gracefully even if this happens.
      */
-    @RunWith(Parameterized.class)
     public static class ExceptionalCasesTest {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStoreSegmentValueFormatterTest.java]: Could you please extract all test cases?

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStoreSegmentValueFormatterTest.java [hunk 18]
```diff
@@ -18,61 +18,49 @@
...
+import org.junit.jupiter.params.provider.MethodSource;
 
-@RunWith(Enclosed.class)
 public class RocksDBVersionedStoreSegmentValueFormatterTest {
 
     /**
      * Non-exceptional scenarios which are expected to occur during regular store operation.
      */
-    @RunWith(Parameterized.class)
     public static class ExpectedCasesTest {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStoreSegmentValueFormatterTest.java]: Could you please extract all test cases?

File: streams/src/test/java/org/apache/kafka/streams/state/internals/TimestampedKeyValueStoreBuilderTest.java [hunk 19]
```diff
@@ -46,28 +48,33 @@ public class TimestampedKeyValueStoreBuilderTest {
...
+    private void setUpWithoutInner() {
         when(supplier.name()).thenReturn("name");
         when(supplier.metricsScope()).thenReturn("metricScope");
 
         builder = new TimestampedKeyValueStoreBuilder<>(
-            supplier,
-            Serdes.String(),
-            Serdes.String(),
-            new MockTime()
+                supplier,
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/TimestampedKeyValueStoreBuilderTest.java]: please revert those unrelated changes

File: streams/src/test/java/org/apache/kafka/streams/state/internals/VersionedKeyValueStoreBuilderTest.java [hunk 20]
```diff
@@ -48,29 +50,34 @@ public class VersionedKeyValueStoreBuilderTest {
...
+    private void setUpWithoutInner() {
         when(supplier.name()).thenReturn(STORE_NAME);
         when(supplier.metricsScope()).thenReturn(METRICS_SCOPE);
 
         builder = new VersionedKeyValueStoreBuilder<>(
-            supplier,
-            Serdes.String(),
-            Serdes.String(),
-            new MockTime()
+                supplier,
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/VersionedKeyValueStoreBuilderTest.java]: please revert those unrelated changes

**Review by chia7712 — COMMENTED** [review]
@m1a2st thanks for updated PR

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStoreSegmentValueFormatterTest.java (hunk 18 above)
Comment (m1a2st) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStoreSegmentValueFormatterTest.java]: @chia7712, Thanks for your comment, I don't really understand that I should extract which test case? Could you talk more details, Thanks for your patience.

**Comment by chia7712** [comment]
@m1a2st Could you please update the jira number? It should be KAFKA-15623, right?

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStoreSegmentValueFormatterTest.java (hunk 18 above)
Comment (chia7712) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStoreSegmentValueFormatterTest.java]: I means - please do a bit refactor to avoid nested tests

**Comment by m1a2st** [comment]
@chia7712, Thanks for your comments, PTAL

File: streams/src/test/java/org/apache/kafka/streams/state/internals/InMemorySessionStoreTest.java [hunk 21]
```diff
@@ -22,15 +22,15 @@
...
 import java.util.HashSet;
 
 import static java.time.Duration.ofMillis;
 import static org.apache.kafka.test.StreamsTestUtils.valuesToSet;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
 
 public class InMemorySessionStoreTest extends AbstractSessionBytesStoreTest {
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/InMemorySessionStoreTest.java:31]:
```java
    @Override
    <K, V> SessionStore<K, V> buildSessionStore(final long retentionPeriod,
                                                 final Serde<K> keySerde,
                                                 final Serde<V> valueSerde) {
        return Stores.sessionStoreBuilder(
            Stores.inMemorySessionStore(
                STORE_NAME,
                ofMillis(retentionPeriod)),
            keySerde,
            valueSerde).build();
    }
```

Could you merge `InMemorySessionStoreTest#buildSessionStore` and `AbstractRocksDBSessionStoreTest#buildSessionStore` into `AbstractSessionBytesStoreTest#buildSessionStore` ?

File: streams/src/test/java/org/apache/kafka/streams/state/internals/InMemoryWindowStoreTest.java [hunk 22]
```diff
@@ -92,7 +94,7 @@ public void shouldRestore() {
 
     @Test
     public void shouldNotExpireFromOpenIterator() {
-
+        setup();
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/InMemoryWindowStoreTest.java]: why? `setup` has tag `BeforeEach`

File: streams/src/test/java/org/apache/kafka/streams/state/internals/InMemoryWindowStoreTest.java [hunk 23]
```diff
@@ -123,7 +125,7 @@ public void shouldNotExpireFromOpenIterator() {
 
     @Test
     public void testExpiration() {
-
+        setup();
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/InMemoryWindowStoreTest.java]: why? `setup` has tag `BeforeEach`

File: streams/src/test/java/org/apache/kafka/streams/state/internals/InMemoryWindowStoreTest.java [hunk 24]
```diff
@@ -167,6 +169,7 @@ public void testExpiration() {
 
     @Test
     public void shouldMatchPositionAfterPut() {
+        setup();
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/InMemoryWindowStoreTest.java]: why? `setup` has tag `BeforeEach`

File: streams/src/test/java/org/apache/kafka/streams/state/internals/ListValueStoreTest.java [hunk 25]
```diff
@@ -31,48 +31,28 @@
...
-import static org.junit.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
-@RunWith(Parameterized.class)
 public class ListValueStoreTest {
     private enum StoreType { InMemory, RocksDB }
 
-    private final StoreType storeType;
+    private StoreType storeType;
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/ListValueStoreTest.java]: This variable can be moved to local

File: streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest.java [hunk 26]
```diff
@@ -16,59 +16,37 @@
...
 import org.apache.kafka.streams.state.internals.PrefixedSessionKeySchemas.TimeFirstSessionKeySchema;
 import org.apache.kafka.streams.state.internals.PrefixedWindowKeySchemas.KeyFirstWindowKeySchema;
 import org.apache.kafka.streams.state.internals.PrefixedWindowKeySchemas.TimeFirstWindowKeySchema;
 import org.apache.kafka.streams.state.internals.SegmentedBytesStore.KeySchema;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
 
-@RunWith(Parameterized.class)
-public class RocksDBTimeOrderedWindowSegmentedBytesStoreTest
+public abstract class AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest.java]: It seems `AbstractRocksDBTimeOrderedWindowSegmentedBytesStoreTest` can be merged into `AbstractDualSchemaRocksDBSegmentedBytesStoreTest`, right?

**Review by chia7712 — COMMENTED** [review]
@m1a2st thanks for updated PR

**Comment by m1a2st** [comment]
@chia7712, Thanks for your comment. PTAL

File: streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSessionStoreTest.java [hunk 27]
```diff
@@ -0,0 +1,33 @@
...
+package org.apache.kafka.streams.state.internals;
+
+public abstract class AbstractRocksDBSessionStoreTest extends AbstractSessionBytesStoreTest {
+
+    @Override
+    String getStoreName() {
+        return "rocksDB session store";
+    }
+
+    abstract StoreType storeType();
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSessionStoreTest.java]: we don't need extra abstract method, right?

Comment (chia7712) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSessionStoreTest.java]: In fact, `AbstractRocksDBSessionStoreTest` is redundant, since its child can implement `getStoreType`. Also, `getStoreName` can be removed since `AbstractSessionBytesStoreTest` can pick up the name.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractDualSchemaRocksDBSegmentedBytesStoreTest.java [hunk 28]
```diff
@@ -107,7 +109,17 @@ public abstract class AbstractDualSchemaRocksDBSegmentedBytesStoreTest<S extends
     final long segmentInterval = 60_000L;
     final String storeName = "bytes-store";
 
-    @Before
+    abstract boolean hasIndex();
```
Comment (chia7712) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractDualSchemaRocksDBSegmentedBytesStoreTest.java]: do we need this method? it seems the `SchemaType` already declares that by the naming.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSessionStoreTest.java (hunk 27 above)
Comment (chia7712) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSessionStoreTest.java]: `getStoreType` can be renamed to `storeType`

**Review by chia7712 — COMMENTED** [review]
@m1a2st thanks for updated PR

File: streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSessionStoreTest.java (hunk 27 above)
Comment (m1a2st) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSessionStoreTest.java]: @chia7712, Thanks for your comment, I think it' a good idea.

**Comment by m1a2st** [comment]
@chia7712, Thanks for your comment, PTAL

**Review by chia7712 — APPROVED** [review]
LGTM
